reconcile/utils/saasherder.py

```diff
@@ -388,13 +390,41 @@ def _process_template(self, options):
                 return None, None
 
             if "IMAGE_TAG" not in consolidated_parameters:
-                template_parameters = template.get('parameters')
-                if template_parameters is not None:
-                    for template_parameter in template_parameters:
-                        if template_parameter['name'] == 'IMAGE_TAG':
-                            # add IMAGE_TAG only if it is required
-                            image_tag = commit_sha[:hash_length]
-                            consolidated_parameters['IMAGE_TAG'] = image_tag
+                for template_parameter in template.get("parameters", {}):
+                    if template_parameter["name"] == "IMAGE_TAG":
+                        # add IMAGE_TAG only if it is required
+                        image_tag = commit_sha[:hash_length]
+                        consolidated_parameters["IMAGE_TAG"] = image_tag
+
+            # Do this in a separate loop, since it relies on IMAGE_TAG already
+            # being calculated.
+            # This should almost never already be set in
+            # consolidated_parameters, but just in case...
+            if "REPO_DIGEST" not in consolidated_parameters:
```

Comment:
can we use "IMAGE_DIGEST" instead?

Reply:
I chose `REPO_DIGEST` to match how `docker image inspect` names the corresponding field (`RepoDigests`, a list) in their JSON output. (I'm assuming that's short for "Repository URI plus image digest".)

Reply:
good thing we are not using docker anymore?

Reply:
Okay, but I agree with them on this point: The "image digest" is the `sha256:{hash}` part, not the whole URI. So IMO `IMAGE_DIGEST` isn't the correct name. And `IMAGE_URI_BY_DIGEST` gets a bit mealy, especially with python line width restrictions enforced at this level of indent :)

Reply:
Fair enough